### PR TITLE
Add ammo itemgroups

### PR DESCRIPTION
**Pull request by mugling**
Provides itemgroups for ammo to match those for guns and magazines

### CODE REVIEW

**Comment by mugling** [comment]
I was intending to work on the gunstores but got stuck with #15318 so this is the standalone changes that can go as a separate PR

File: data/json/itemgroups/ammo.json [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+       [ "9mmfmj", 100 ]
+    ]
+  },
+  {
+    "type" : "item_group",
+    "id" : "ammo_pistol_obscure",
+    "//" : "Imported or otherwise very obscure pistol ammmo.",
+    "items": [
+      [ "9x18mm", 100 ],
+      [ "9x18mmfmj", 50 ]
```
Comment (Malkeus) [inline data/json/itemgroups/ammo.json]: Damn the commas!

File: data/json/itemgroups/ammo.json [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+       [ "reloaded_45_jhp", 100 ],
+       [ "reloaded_45_super", 40 ],
+       [ "reloaded_46mm", 20 ],
+       [ "reloaded_57mm", 50 ],
+       [ "reloaded_9mm", 300 ],
+       [ "reloaded_9mmfmj", 100 ],
+       [ "reloaded_9mmP", 80 ],
+       [ "reloaded_9mmP2", 40 ]
+       [ "reloaded_9x18mm", 20 ],
+       [ "reloaded_9x18mmfmj", 10 ]
```
Comment (Malkeus) [inline data/json/itemgroups/ammo.json]: Those hateful commas gonna hate.

Comment (mugling) [inline data/json/itemgroups/ammo.json]: Back to this later, just working on another branch

**Pull request merged**